### ISSUE TITLE
How to reuse Docker images across blocks using cache

### ISSUE THREAD

**Issue by markoa** [semaphoreci/docs#173]
Let's complete #172 before starting work on this one. This would be a new take which would supersede #74, so please create a new pull request.

Scenario: I build Docker images based on my code in one block and then want to carry them to the next blocks as fast as possible. After we're done with testing phases and the image is promoted — let's say I've published my service/app to a registry — the image(s) are no longer needed in cache. So we can delete them from cache at the end of the workflow. This approach keeps cache use at a normal level.

**Comment by markoa** [comment]
Using cache is suboptimal comparing to registry due to large image sizes.
